lib/std/os/linux.zig

```diff
@@ -2141,40 +2166,40 @@ pub fn process_vm_writev(pid: pid_t, local: []const iovec_const, remote: []const
 }
 
 pub fn fadvise(fd: fd_t, offset: i64, len: i64, advice: usize) usize {
-    if (comptime builtin.cpu.arch.isMIPS()) {
-        // MIPS requires a 7 argument syscall
+    if (comptime native_arch.isARM() or native_arch.isPPC()) {
+        // These architectures reorder the arguments so that a register is not skipped to align the
```

Comment:
I can try. This was deep in glibc innards.

Reply:
Believe it's this bit: https://github.com/bminor/glibc/blob/910aae6e5a2196938fc30fa54dd1e96f16774ce7/sysdeps/unix/sysv/linux/posix_fadvise.c#L27-L28

And musl: https://git.musl-libc.org/cgit/musl/tree/src/fcntl/posix_fadvise.c#n7

I spent a good day piecing together this `fadvise` ABI nonsense from kernel, glibc, and musl sources. It's a mess. I've happily purged it from my memory since. I'm pretty sure I got it right, though.

Reply:
Gotcha, thanks for digging it up. We'll have it here for reference if ever needed again.